src/WebJobs.Script/Extensions/ExceptionExtensions.cs

```diff
@@ -14,21 +14,16 @@ public static bool IsFatal(this Exception exception)
         {
             while (exception != null)
             {
-                if ((exception is OutOfMemoryException && !(exception is InsufficientMemoryException)) ||
-                    exception is ThreadAbortException ||
-                    exception is AccessViolationException ||
-                    exception is SEHException ||
-                    exception is StackOverflowException)
+                if (exception
+                    is (OutOfMemoryException and not InsufficientMemoryException)
+                    or ThreadAbortException
+                    or AccessViolationException
+                    or SEHException
+                    or StackOverflowException)
                 {
                     return true;
                 }
 
-                if (exception is TypeInitializationException &&
-                    exception is TargetInvocationException)
-                {
-                    break;
-                }
-
```

Comment:
This is the key change. Removed this check as the logic bug here made this `break` unreachable.

The host has been running for years without this logic, so I didn't want to change it to start considering these types.